docs/data-sources/collection.md

```diff
@@ -59,3 +59,4 @@ data "pinecone_collection" "test" {
 - `id` (String) Collection identifier
 - `size` (Number) The size of the collection in bytes.
 - `status` (String) The status of the collection.
+- `vector_count` (Number) The number of records stored in the collection.
```

Comment:
All the changes under `/docs/` are generated using `go generate ./...` from the root.